runtime/j9vm/jvm.c

```diff
@@ -1369,164 +1354,6 @@ decodeSetting(const char* key, const char* value, const VersionSetting* settings
 	return 0;
 }
 
-/**
- * Attempt loading 'classlib.properties' file, and get Java version info.
- * If the file is found, 'version' and 'shape' values are retrieved
- * and decoded as J2SE_xx and J2SE_SHAPE_xx accordingly.
- * 'J2SE_xx | J2SE_SHAPE_xx' is returned;
- * Otherwise, 0 is returned.
- *
- * @return 'J2SE_xx | J2SE_SHAPE_xx' decoded from 'version' and 'shape' values in 'classlib.properties';
- *         or 0 if otherwise.
- */
-static UDATA
-getVersionFromClasslibPropertiesFile(void)
-{
-	PORT_ACCESS_FROM_PORT(&j9portLibrary);
-	J9StringBuffer *propsPathBuffer = NULL;
-	j9props_file_t propsFile = NULL;
-	UDATA finalVersion = 0;
-
-	propsPathBuffer = jvmBufferCat(propsPathBuffer, jvmBufferData(j9libBuffer));
-	propsPathBuffer = jvmBufferCat(propsPathBuffer, DIR_SEPARATOR_STR "classlib.properties");
-	propsFile = props_file_open(PORTLIB, jvmBufferData(propsPathBuffer), NULL, 0);
-	free(propsPathBuffer);
-	propsPathBuffer = NULL;
-
-	if (NULL != propsFile) {
-		const char *shape = NULL;
-		const char *version = NULL;
-		UDATA decoded = 0;
-		
-		shape = props_file_get(propsFile, "shape");
-		if (NULL == shape) {
-#ifdef DEBUG
-			printf("No 'shape' property in %s\n", jvmBufferData(propsPathBuffer));
-#endif
-			goto bail;
-		}
-
-		version = props_file_get(propsFile, "version");
-		if (NULL == version) {
-#ifdef DEBUG
-			printf("No 'version' property in %s\n", jvmBufferData(propsPathBuffer));
-#endif
-			goto bail;
-		}
-	
-		decoded = decodeSetting("shape", shape, SHAPE_SETTINGS, NUM_SHAPE_SETTINGS);
-		if (0 == decoded) {
-			goto bail;
-		}
-		finalVersion |= decoded;
-	
-		decoded = decodeSetting("version", version, VERSION_SETTINGS, NUM_VERSION_SETTINGS);
```

Comment:
Is there any remaining use of decodeSetting()? Seem the impl can be deleted. Same with VERSION_SETTINGS.

Reply:
Agreed, will remove it.